drivers/timer/stm32_lptim_timer.c

```diff
@@ -84,18 +84,32 @@ static struct k_spinlock lock;
 
 #ifdef CONFIG_STM32_LPTIM_STDBY_TIMER
 
+#if !defined(CONFIG_PM_S2RAM) && !defined(CONFIG_STM32_USE_STOP3)
+#error CONFIG_PM_S2RAM or CONFIG_STM32_USE_STOP3 needs to be selected
+#endif
+
 #define CURRENT_CPU \
 	(COND_CODE_1(CONFIG_SMP, (arch_curr_cpu()->id), (_current_cpu->id)))
 
 #define cycle_t uint32_t
 
+typedef enum {
+	STM32_LPTIM_TIMEOUT_NONE = 0,
+#if defined(CONFIG_PM_S2RAM)
+	STM32_LPTIM_TIMEOUT_STANDBY = 1,
+#endif
+#if defined(CONFIG_STM32_USE_STOP3)
+	STM32_LPTIM_TIMEOUT_STOP3 = 2,
+#endif
```

Comment:
You don't need to make both modes coexisting: S2RAM is only for WBA and WBA doesn't support STOP3

Reply:
In theory the U5 might support suspend to RAM, although the RAM is limited to SRAM2 and 64kB. So, I wanted to make the code more future proof.

Reply:
If this could allow to simplify the code, I'd prefer to not consider it for now. This would also fasten the review process.